src/static/riot/competitions/editor/_participation.tag

```diff
@@ -0,0 +1,53 @@
+<competition-participation>
+    <form class="ui form">
+        <div class="field required">
+            <label>Terms</label>
+            <textarea class="markdown-editor" ref="terms" name="terms"></textarea>
+        </div>
+        <div class="field">
+            <div class="ui checkbox">
+                <input type="checkbox" name="registration_auto_approve" ref="registration_auto_approve" onchange="{form_updated}">
+                <label>Auto approve registration requests
+                    <span data-tooltip="If left unchecked, registration requests must be manually approved by the competition creator or collaborators"
+                          data-inverted=""
+                          data-position="bottom center">
+                    <i class="help icon circle"></i></span>
+                </label>
+            </div>
+        </div>
+    </form>
+
+    <script>
+        let self = this
+
+        self.data = {}
+
+        self.on('mount', () => {
+            self.markdown_editor = create_easyMDE(self.refs.terms)
+
+            $(':input', self.root).not('[type="file"]').not('button').not('[readonly]').each(function (i, field) {
+                this.addEventListener('keyup', self.form_updated)
```

Comment:
do we need to clean up `addEventListeners`?

Reply:
what to you mean "clean up"?

Reply:
_Disclaimer: may not be worth spending much time on, since we're not a long running single page app, we have page refreshes._

By "clean up" I mean implement some kind of `removeEventListener` when we're unmounted/what-have-you. The current code _may_ add listeners without them being garbage collected over and over and over again -- but, I may be misunderstanding how things are setup. If we're hiding the element, maybe these listeners aren't destroyed properly? If the element is destroyed, the event listener should be destroyed with it?

https://stackoverflow.com/questions/446892/how-to-find-event-listeners-on-a-dom-node-when-debugging-or-from-the-javascript

Looks like we could call `getEventListeners(dom element)` to check to see if things aren't cleaned up? Again, probably not worth the time to get into

Reply:
Just using the event listeners section of chrome console, looks like these event listeners go away when you navigate away from the page, but there are a lot of them on the comp editor page since all the tags are mounted together, and then just hidden in tabs. Could maybe clean this up so there aren't like 30 event listeners or whatever, but probably not necessary?

Reply:
Yeah, probably not a huge worry now -- we'll keep an eye on it every time we manually do some `addEventListener` jazz